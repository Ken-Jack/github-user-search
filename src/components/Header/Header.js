// @flow

import React from 'react';
import SearchForm from 'components/SearchForm/SearchForm';
import Container from 'components/Container/Container';
import Logo from 'components/Logo/Logo';
import {connect} from 'react-fela';

type Props = {
  onSubmit: Function,
  searchTerm: string,
  styles: Object,
};

function Header({onSubmit, searchTerm, styles}: Props) {
  return (
    <Container rootStyle={styles.Header_container}>
      <div
        className={styles.Header_root}
      >
        <div className={styles.Header_logo}>
          <Logo />
        </div>
        <div className={styles.Header_form}>
          <SearchForm
            onSubmit={onSubmit}
            initialInputValue={searchTerm}
          />
        </div>
      </div>
    </Container>
  );
}

const MQ = '@media (min-width: 600px)';

const styles = {
  Header_root: () => ({
    paddingTop: '15px',
    paddingBottom: '15px',

    [MQ]: {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
  }),

  Header_logo: () => ({
    flexGrow: 2,
  }),

  Header_container: () => ({
    backgroundColor: '#f4f4f4',
    borderBottom: '1px solid #aaa',
    boxShadow: '0px 0px 2px 1px rgba(0, 0, 0, .2)',
  }),

  Header_form: () => ({
    marginTop: '20px',

    [MQ]: {
      margin: 0,
      flexGrow: 1,
    },
  }),
};

export default connect(styles)(Header);
