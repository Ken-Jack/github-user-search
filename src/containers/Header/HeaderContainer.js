// @flow

import React, {
  Component,
} from 'react';
import Header from 'components/Header/Header';
import connect from './connect';

type Props = {
  onSubmit: Function,
  searchQuery: string,
  searchTerm: string,
  searchForUser: Function,
};

export class HeaderContainer extends Component {

  static defaultProps = {
    searchTerm: '',
  };

  constructor(props: Props) {
    super(props);
    this.handleSearchUser(props.searchQuery);
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.searchQuery !== this.props.searchQuery) {
      this.handleSearchUser(nextProps.searchQuery);
    }
  }

  handleSearchUser(searchQuery: string): void {
    if (!searchQuery) {return;}
    this.props.searchForUser(searchQuery);
  }

  render() {
    const {
      onSubmit,
      searchTerm,
    } = this.props;

    return (
      <Header
        onSubmit={onSubmit}
        searchTerm={searchTerm}
      />
    );
  }

}

export default connect(HeaderContainer);
