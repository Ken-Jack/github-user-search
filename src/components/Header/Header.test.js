import React from 'react';
import {
  StaticRouter as Router,
} from 'react-router-dom';
import felaSnapshot from 'test-util/fela-snapshot';

import Header from './Header';

describe('Component: Header', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Router context={{}}>
        <Header onSubmit={jest.fn()} searchTerm="test">
          <div>inner component</div>
        </Header>
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
