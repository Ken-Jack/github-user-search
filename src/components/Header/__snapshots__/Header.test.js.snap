// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render 1`] = `
Object {
  "component": <div
    className="m g h i"
  >
    <div
      className="n o"
    >
      <div
        className="a b c d e"
      >
        <div
          className="f"
        >
          <div>
            <h1
              className="t"
            >
              <a
                className="u v w"
                href="/"
                onClick={[Function]}
              >
                <svg />
                GitHub User Search
              </a>
            </h1>
            <p
              className="x y"
            >
              Browse users and their profiles via 
              <a
                href="https://developer.github.com/v3/"
              >
                the GitHub API
              </a>
            </p>
          </div>
        </div>
        <div
          className="j k l"
        >
          <form
            className="z"
            onSubmit={[Function]}
          >
            <label
              className="u-hiddenVisually"
              htmlFor="searchInput"
            >
              Search for a user
            </label>
            <input
              className="ab ac ae af ag"
              id="searchInput"
              onChange={[Function]}
              placeholder="Search for a user, e.g. simonsmith"
              type="text"
              value="test"
            />
            <button
              className="ah ai aj ak"
              type="submit"
            >
              <svg />
              <span
                className="u-hiddenVisually"
              >
                Submit
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "styles": "
    .a{padding-top:15px}
    .b{padding-bottom:15px}
    .f{flex-grow:2}
    .g{background-color:#f4f4f4}
    .h{border-bottom:1px solid #aaa}
    .i{box-shadow:0px 0px 2px 1px rgba(0, 0, 0, .2)}
    .j{margin-top:20px}
    .m{margin:0}
    .n{margin:0 auto}
    .o{max-width:1020}
    .p{padding-left:10}
    .q{padding-right:10}
    .s{margin-right:8px}
    .t{font-size:25px}
    .u{display:flex}
    .v{text-decoration:none}
    .w{color:inherit}
    .x{margin-top:5px}
    .y{font-size:14px}
    .z{position:relative}
    .ab{padding:8px}
    .ac{padding-right:40px}
    .ae{width:100%}
    .af{border:1px solid #bbb}
    .ag{box-shadow:inset 0 2px 2px rgba(0, 0, 0, .1);}
    .ah{padding:3px 14px}
    .ai{position:absolute}
    .aj{top:4px}
    .ak{right:0}@media (min-width: 600px){
    .c{display:flex}
    .d{justify-content:space-between}
    .e{align-items:center}
    .k{margin:0}
    .l{flex-grow:1}}@media (min-width: 1050px){
    .r{padding:0}}",
}
`;
